node/router/src/helpers/cache.rs

```diff
@@ -221,9 +226,36 @@ impl<N: Network> Cache<N> {
         timestamps.len()
     }
 
+    /// Insert a new timestamp for the given key, returning the number of recent entries and whether the request is higher than the last.
+    fn retain_incrementing_requests_and_insert<K: Eq + Hash + Clone>(
+        map: &RwLock<HashMap<K, VecDeque<(OffsetDateTime, BlockRequest)>>>,
+        key: K,
+        request: BlockRequest,
+        interval_in_secs: i64,
+    ) -> (usize, bool) {
+        // Fetch the current timestamp.
+        let now = OffsetDateTime::now_utc();
+
+        let mut map_write = map.write();
+        // Load the entry for the key.
+        let timestamps = map_write.entry(key).or_default();
+        // Retain only the timestamps that are within the recent interval.
+        while timestamps.front().map_or(false, |(t, _)| now - *t > Duration::seconds(interval_in_secs)) {
+            timestamps.pop_front();
+        }
+        // Check whether the request is higher than the last.
+        let incrementing = timestamps.back().map(|(_, last)| request.start_height > last.end_height).unwrap_or(true);
```

Comment:
```suggestion
        let incrementing = timestamps.back().map(|(_, last)| request.start_height >= last.end_height).unwrap_or(true);
```

Should it be a `>=` because the `end_height` is exclusive whereas the `start_height` is inclusive.

Reply:
255a459a0